src/unistd.rs

```diff
@@ -402,12 +476,39 @@ pub fn sethostname(name: &[u8]) -> Result<()> {
     Errno::result(res).map(drop)
 }
 
-pub fn gethostname(name: &mut [u8]) -> Result<()> {
-    let ptr = name.as_mut_ptr() as *mut c_char;
-    let len = name.len() as size_t;
+/// Get the host name and store it int the provided buffer, returning the actual
+/// length stored in the buffer (see
+/// [gethostname(2)](http://man7.org/linux/man-pages/man2/gethostname.2.html)).
```

Comment:
Ah, yes.  This needs to be updated.  I went through a few iterations and ended up changing how this worked.